ks_patches/innreise_duplicates.js

```diff
@@ -0,0 +1,141 @@
+                return newTeiId;
+            });
+        });
+    });
+}
+
+function updateAndAddInnreise(selectedTei, existingTeiId, isExisting, optionSets, attributesById, orgId, teiService, enrollmentService, eventFactory) {
+    return teiService.getWithProgramData(existingTeiId, isExisting ? INNREISE_PROGRAM_CODE : DUPLIKAT_PROGRAM_CODE, optionSets, attributesById).then(existingTei => {
+        try {
+            return updateProfile(selectedTei, existingTei, optionSets, attributesById, teiService).then(() => {
+                if (isEnrolledInInnreise(existingTei)) {
+                    return addInnreiseEvents(selectedTei, existingTei, innreiseEnrollmentCode(existingTei), eventFactory);
+                } else {
+                    addInnreiseEnrollment(selectedTei, existingTei, orgId, enrollmentService).then((response) => {
+                        var enrollmentId = response.response.importSummaries[0].reference;
+                        return addInnreiseEvents(selectedTei, existingTei, enrollmentId, eventFactory);
+                    });
+                }
+            });
+        } catch (err) {
+            console.log(err)
+            throw err;
+        }
+    });
+}
+
+
+function addInnreiseEnrollment(selectedTei, existingTei, orgId, enrollmentService) {
+    var enrollment = createEnrollment(existingTei.trackedEntityInstance, orgId);
+    return enrollmentService.enroll(enrollment);
+}
+
+function addInnreiseEvents(selectedTei, existingTei, enrollmentId, eventFactory) {
+
+    selectedTei.enrollments.forEach(enrollment => {
+        if (enrollment.program === DUPLICAT_INNREISE_PROGRAM_CODE) {
+            enrollment.events.forEach(event => {
+                try {
+                    var newEvent = angular.copy(event);
+                    newEvent.program = INNREISE_PROGRAM_CODE;
+                    newEvent.programStage = duplicateToInnreiseEventCode(event.programStage);
+                    var id = Array.isArray(enrollmentId) ? enrollmentId[0] : enrollmentId;  // Needed to avoid "Event date is required."-error
+                    newEvent.enrollment = id;
+                    if (newEvent.notes) {
+                        newEvent.notes = newEvent.notes.map(note => {
+                            delete note.note; // Cannot reuse note ID
+                            return note;
+                        });
+                    }
+                    delete newEvent.event; // Cannot reuse event ID
+                    if (newEvent.programStage) {
+                        eventFactory.create(newEvent);
+                    }
+                } catch (err) {
+                    console.log(err);
+                    throw err;
+                }
+            });
+        }
+    });
+}
+
+function updateProfile(selectedTei, existingTei, optionSets, attributesById, teiService) {
+    selectedTei.attributes.forEach((attribute) => {
+        if (!existingTei.attributes.some((att) => att.code === attribute.code)) {
```

Comment:
It will be safer to use the uid when comparing, as the code might be blank.

Reply:
changed to `att.id === attribute.id`